.github/workflows/auto-pr-submodules.yml

```diff
@@ -0,0 +1,38 @@
+# This is a basic workflow to help you get started with Actions
+name: Update Submodules and Submit a PR
+
+# Controls when the action will run.
+on:
+  # Triggers the workflow on push  only for the actions-test-branch branches
+  push:
+    branches:
+      - master
+    # Runs every day at midnight UTC
+    schedule:
+      - cron:  '0 0 * * *'
+
+jobs:
+  createPullRequest:
+    if: ${{ contains(github.ref, 'master') }}
```

Comment:
I think that `if: contains(github.ref, 'master')` would work too. No need fo `${{}}`.

Moreover, if the branch is already limited above, is this condition necessary here?

Reply:
Yeah, it works.

The `schedule:` cron will run on any branch regardless of the `push: branches` list, am I correct?

Reply:
The `schedule` cron is only executed from the default branch (`master` in this case). Hence, by definition/design, it's not possible to run schedules on any other branch. That's not a problem.

I was going to suggest adding a `workflow_dispatch` event too, so we can trigger the workflow manually through the web UI. In that case, triggering for any branch would be possible, and this condition would be pertinent. However, I'm unsure about the manual dispatch being useful in this specific workflow. wdyt?

Reply:
@umarcor: I agree workflow_dispatch can prove useful in some cases.